pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -219,13 +250,74 @@ public double[] transformToDoubleValuesSV(ProjectionBlock projectionBlock) {
           String[] stringValues = transformToStringValuesSV(projectionBlock);
           ArrayCopyUtils.copy(stringValues, _doubleValuesSV, length);
           break;
+        case BYTES:
+          if (getResultMetadata().getDataType().equals(DataType.BIG_DECIMAL)) {
+            BigDecimal[] bigDecimalValues = transformToBigDecimalValuesSV(projectionBlock);
+            ArrayCopyUtils.copy(bigDecimalValues, _doubleValuesSV, length);
+            break;
+          }
+          throw new IllegalStateException();
         default:
           throw new IllegalStateException();
       }
     }
     return _doubleValuesSV;
   }
 
+  // Note 1: Why need transformToBigDecimalValuesSV(projectionBlock) while there is no one for TIMESTAMP data type?
+  // This is because TIMESTAMP's underlying stored type is LONG and we already have a method for LONG that can convert
+  // to other numeric data types (check: transformToLongValuesSV(projectionBlock) method).
+  // However, BIG_DECIMAL's underlying stored type is BYTES and transformToBytesValuesSV(projectionBlock) handles only
+  // the conversion from STRING (check: transformToBytesValuesSV(projectionBlock) method).
+  //
+  // Note 2: An alternative way of transforming projectionBlock into BigDecimal[] values is to call
+  // transformToBytesValuesSV(projectionBlock) method that return byte[] values and then transform byte[] to
+  // BigDecimal[]. However, this requires three operations (vs. two in this method):
+  // 1- transformTo[Type]ValuesSV(projectionBlock) based on the underlying stored type.
+  // 2- ArrayCopyUtils.copy([Type]Values, bytesValues, length).
+  // 3- ArrayCopyUtils.copy(bytesValues, bigDecimals, length).
+  @Override
+  public BigDecimal[] transformToBigDecimalValuesSV(ProjectionBlock projectionBlock) {
```

Comment:
Let's not add this method for now. We might not want to support implicit cast to/from big decimal to other data types (this is not the SQL semantic as well). Users can cast explicitly using transform function